spacy/pipeline/coref.py

```diff
@@ -218,6 +218,13 @@ def update(
         total_loss = 0
 
         for eg in examples:
+            if eg.x.text != eg.y.text:
```

Comment:
Not sure how expensive is this in `Python` for longer documents. Is this something we must do?

Reply:
It is something that you have to do because otherwise you can't know whether you can actually compare the character offsets. If speed turns out to be an issue, it could be cached in `Example` instead, maybe?

Reply:
I believe that Python caches the hash for strings and only compares that, but it's good to keep in mind.